packages/squiggle-lang/src/rescript/Reducer/Reducer_ErrorValue.res

```diff
@@ -0,0 +1,26 @@
+type errorValue =
+  | REArrayIndexNotFound(string, int)
+  | REFunctionExpected(string)
+  | REJs(option<string>, option<string>) // Javascript Exception
+  | RERecordPropertyNotFound(string, string)
+  | RETodo(string) // To do
+
+let showError = err =>
+  switch err {
+  | REArrayIndexNotFound(msg, index) => `${msg}: ${Js.String.make(index)}`
+  | REFunctionExpected(msg) => `Function expected: ${msg}`
+  | REJs(omsg, oname) => {
+      let answer = "JS Exception:"
+      let answer = switch oname {
+      | Some(name) => `${answer} ${name}`
+      | _ => answer
+      }
+      let answer = switch omsg {
+      | Some(msg) => `${answer}: ${msg}`
+      | _ => answer
+      }
```

Comment:
I feel like the above could be compressed nicer, maybe like,
```
`JS Exception: ${oname |> E.O.default("")} ${omsg |> E.O.fmap(o => `:${o}`)`
```

Reply:
Could be. More haskelly but less readable...
Also... I am delaying all module E dependencies until after
   module E is refactored and the other parser is removed.
I will keep warning if we create a module E dependency now that will create a series of merging problems in the weeks to follow.

Reply:
RESOLVED. Another task